documentation/asciidoc/computers/linux_kernel/building.adoc

```diff
@@ -1,342 +1,462 @@
 [[building]]
-== Building the kernel
+== Build the kernel
 
-The default compilers and linkers that come with an OS are configured to build executables to run on that OS - they are native tools - but that doesn't have to be the case. A cross-compiler is configured to build code for a target other than the one running the build process, and using it is called cross-compilation.
+The default compilers and linkers distributed with an OS are configured to build executables to run on that OS. **Cross-compilation** is the process of building code for a target other than the one running the build process.
 
-Cross-compilation of the Raspberry Pi kernel is useful for two reasons: it allows a 64-bit kernel to be built using a 32-bit OS, and vice versa. It also means that even a modest laptop can cross-compile a Raspberry Pi kernel significantly faster than the Raspberry Pi itself.
+Cross-compilation of the Raspberry Pi kernel allows you to build a 64-bit kernel from a 32-bit OS, and vice versa. Alternatively, you can cross-compile a 32-bit or 64-bit Raspberry Pi kernel from a device other than a Raspberry Pi.
 
-The instructions below are divided into native builds and cross-compilation. Choose the section appropriate for your situation; although there are many common steps between the two, there are also some important differences.
+[NOTE]
+====
+The 32-bit distribution of Raspberry Pi OS on a Raspberry Pi 4B, 5, 400, Compute Module 4, or Compute Module 4S uses a 32-bit userland, but a _64-bit kernel_. To build a 32-bit kernel, set `ARCH=arm`. To boot a 32-bit kernel, set `arm_64bit=0` in `config.txt`.
+====
 
-=== Building the kernel locally
+The instructions below are divided into native builds and cross-compilation. Choose the section appropriate for your situation; although the two processes share many steps, there are also some important differences.
```

Comment:
These sections are **very** long, so probably worth adding in anchor-links to the first sentence here? (i.e. to the "Native builds" and "Cross-compiled builds" sections)